src/titlecase/full.rs

```diff
@@ -0,0 +1,428 @@
+use core::char::ToLowercase;
+use core::fmt;
+use core::iter::FusedIterator;
+use core::ops::Range;
+
+use bstr::ByteSlice;
+
+pub use crate::unicode::Titlecase as TitlecaseForChar;
+use crate::unicode::ToTitlecase;
+
+#[derive(Clone, Debug)]
+enum ToCase {
+    ToLowercase(ToLowercase),
+    ToTitlecase(ToTitlecase),
+}
+
+impl Iterator for ToCase {
+    type Item = char;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            ToCase::ToLowercase(iter) => iter.next(),
+            ToCase::ToTitlecase(iter) => iter.next(),
+        }
+    }
+}
+
+#[derive(Clone)]
+#[must_use = "Titlecase is a Iterator and must be used"]
+pub struct Titlecase<'a> {
+    slice: &'a [u8],
+    next_bytes: [u8; 4],
+    next_range: Range<usize>,
+    case_iter: Option<ToCase>,
+    beginning: bool,
```

Comment:
`first` instead of `beginning` to be consistent with the ascii implementation?

Reply:
Thanks I didn't notice that one! Fixed in 01d0ac0.